### PR TITLE
Dev: Azure Automation cmdlets 

### PR DESCRIPTION
**Pull request by safeermohammed**
Added Remove-AzureRMAutomationConnectionType cmdlet
Added test for cmdlet
Bug fixes in DSC compilation job and few fixes

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@safeermohammed__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (safeerm). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: src/ResourceManager/Automation/Commands.Automation/Cmdlet/StartAzureAutomationDscCompilationJob.cs [hunk 1]
```diff
@@ -25,7 +28,7 @@ namespace Microsoft.Azure.Commands.Automation.Cmdlet
     /// <summary>
     /// starts azure automation compilation job
     /// </summary>
-    [Cmdlet(VerbsLifecycle.Start, "AzureRmAutomationDscCompilationJob")]
+    [Cmdlet(VerbsLifecycle.Start, "AzureRMAutomationDscCompilationJob")]
```
Comment (stankovski) [inline src/ResourceManager/Automation/Commands.Automation/Cmdlet/StartAzureAutomationDscCompilationJob.cs]: Please rename back to AzureRm

File: src/ResourceManager/Automation/Commands.Automation.Test/UnitTests/RemoveAzureAutomationConnectionTypeTest.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+using Microsoft.Azure.Commands.Automation.Common;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.WindowsAzure.Commands.Common.Test.Mocks;
+using Microsoft.WindowsAzure.Commands.ScenarioTest;
+using Microsoft.WindowsAzure.Commands.Test.Utilities.Common;
+using Moq;
+
+namespace Microsoft.Azure.Commands.ResourceManager.Automation.Test.UnitTests
+{
+    [TestClass]
```
Comment (stankovski) [inline src/ResourceManager/Automation/Commands.Automation.Test/UnitTests/RemoveAzureAutomationConnectionTypeTest.cs]: Please convert to xunit. We do not run any MSTest tests as part of Pull Request validation.
(https://github.com/Azure/azure-powershell/blob/dev/AzurePowershell.Test.targets#L96)

File: src/ResourceManager/Automation/Commands.Automation/Cmdlet/ImportAzureAutomationDscNodeConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+using Microsoft.Azure.Commands.Automation.Common;
+using Microsoft.Azure.Commands.Automation.Model;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+
+namespace Microsoft.Azure.Commands.Automation.Cmdlet
+{
+    /// <summary>
+    /// Imports dsc node configuration script
+    /// </summary>
+    [Cmdlet(VerbsData.Import, "AzureRMAutomationDscNodeConfiguration")]
```
Comment (stankovski) [inline src/ResourceManager/Automation/Commands.Automation/Cmdlet/ImportAzureAutomationDscNodeConfiguration.cs]: Rename to AzureRmXXXX

File: src/ResourceManager/Automation/Commands.Automation/Cmdlet/RemoveAzureAutomationConnectionType.cs [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+using System.Security.Permissions;
+using Microsoft.Azure.Commands.Automation.Common;
+using Microsoft.Azure.Commands.Automation.Properties;
+
+namespace Microsoft.Azure.Commands.Automation.Cmdlet
+{
+    /// <summary>
+    /// Removes a Connection type for automation.
+    /// </summary>
+    [Cmdlet(VerbsCommon.Remove, "AzureRMAutomationConnectionType", DefaultParameterSetName = AutomationCmdletParameterSets.ByName)]
```
Comment (stankovski) [inline src/ResourceManager/Automation/Commands.Automation/Cmdlet/RemoveAzureAutomationConnectionType.cs]: Rename to AzureRmXXXX

File: src/ServiceManagement/Automation/Commands.Automation/Cmdlet/RemoveAzureAutomationConnectionType.cs [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+using Microsoft.Azure.Commands.Automation.Common;
+using Microsoft.Azure.Commands.Automation.Properties;
+
+namespace Microsoft.Azure.Commands.Automation.Cmdlet
+{
+    /// <summary>
+    /// Removes a Connection type for automation.
+    /// </summary>
+    [Cmdlet(VerbsCommon.Remove, "AzureAutomationConnectionType", DefaultParameterSetName = AutomationCmdletParameterSets.ByName)]
+    public class RemoveAzureAutomationConnectionType : AzureAutomationBaseCmdlet
```
Comment (stankovski) [inline src/ServiceManagement/Automation/Commands.Automation/Cmdlet/RemoveAzureAutomationConnectionType.cs]: Rename to AzureRmXXXX

Comment (safeermohammed) [inline src/ServiceManagement/Automation/Commands.Automation/Cmdlet/RemoveAzureAutomationConnectionType.cs]: fixed

File: src/ResourceManager/Automation/Commands.Automation.Test/UnitTests/RemoveAzureAutomationConnectionTypeTest.cs (hunk 2 above)
Comment (safeermohammed) [inline src/ResourceManager/Automation/Commands.Automation.Test/UnitTests/RemoveAzureAutomationConnectionTypeTest.cs]: These are unit tests only. Will take up in next release cycle to move them to xunit.  
Our Scenario tests are xunit based.

File: src/ServiceManagement/Automation/Commands.Automation/Cmdlet/RemoveAzureAutomationConnectionType.cs (hunk 5 above)
Comment (stankovski) [inline src/ServiceManagement/Automation/Commands.Automation/Cmdlet/RemoveAzureAutomationConnectionType.cs]: Where? Line 27 still shows `AzureAutomationConnectionType`

Comment (safeermohammed) [inline src/ServiceManagement/Automation/Commands.Automation/Cmdlet/RemoveAzureAutomationConnectionType.cs]: This is service management cmdlet, 
I fixed under ARM cmdlet.  We want servicemanagement cmdlet as is right?

Comment (stankovski) [inline src/ServiceManagement/Automation/Commands.Automation/Cmdlet/RemoveAzureAutomationConnectionType.cs]: Got you. Didn't notice it was Service Management,

**Pull request merged**